src/Data/Avro/Decode/Lazy.hs

```diff
@@ -0,0 +1,283 @@
+
+-- | Decodes the container as a lazy list of values of the requested type.
+--
+-- The schema for the requested type will be de-conflicted with the schema
+-- embedded with the container.
+--
+-- The content of the container is returned as a list ob "blocks" of values
+-- inside this container, so the notion of blocks in the container is preserved.
+-- Since decoding is lazy it should be safe to concat these values into one lazy list.
+--
+-- The "outer" error represents the error in opening the container itself
+-- (including problems like reading schemas embedded into the container.)
+--
+-- The "inner" errors represent problems in decoding individual values.
+--
+-- Note that this function will not stop decoding at the first occurance of the "inner"
+-- error, and will continue attempting decoding values, so it is possible to
+-- get 'Right' after 'Left'. It is up to the user to decide whether it is correct or not to
+-- continue after errors (most likely it will not be correct).
+--
+-- 'decodeContainer' function makes a choice to stop after the first error.
+decodeContainer' :: forall a. FromAvro a => BL.ByteString -> Either String [[Either String a]]
+decodeContainer' = decodeContainerWithSchema' (untag (schema :: Tagged a Schema))
+
+-- | Same as 'decodeContainer' but uses provided schema as a reader schema for the container
+-- instead of the schema obtained from the type of 'a'.
+--
+-- It is up to the user to make sure that the provided schema is compatible with 'a'
+-- and with the container's writer schema.
+decodeContainerWithSchema :: FromAvro a => Schema -> BL.ByteString -> [Either String a]
+decodeContainerWithSchema s bs =
+  either (\err -> [Left err]) concat (decodeContainerWithSchema' s bs)
+
+-- | Same as 'decodeContainer'' but uses provided schema as a reader schema for the container
+-- instead of the schema obtained from the type of 'a'.
+--
+-- It is up to the user to make sure that the provided schema is compatible with 'a'
+-- and with the container's writer schema.
+decodeContainerWithSchema' :: FromAvro a => Schema -> BL.ByteString -> Either String [[Either String a]]
+decodeContainerWithSchema' readerSchema bs = do
+  (writerSchema, vals) <- getContainerValues bs
+  pure $ (fmap . fmap) (convertValue writerSchema) vals
+  where
+    convertValue w v = toStrictValue (C.deconflict w readerSchema v) >>= (resultToEither . fromAvro)
+
+-- |Decode bytes into a 'Value' as described by Schema.
+decodeAvro :: Schema -> BL.ByteString -> T.LazyValue Type
+decodeAvro s = snd . getAvroOf s
+{-# INLINABLE decodeAvro #-}
+
+-- | Decodes the container into a list of blocks of raw Avro values.
+--
+-- The content of the container is returned as a list ob "blocks" of values
+-- inside this container, so the notion of blocks in the container is preserved.
+-- Since decoding is lazy it should be safe to concat these values into one lazy list.
+--
+-- Each 'LazyyValue' can be an `Error' and this function doesn't make any attempts
```

Comment:
typo "Lazyy"

Reply:
`Lazyy` means "very lazy" :)
Thanks for reviewing! :)